examples/3d/debug_gltf.rs

```diff
@@ -0,0 +1,112 @@
+
+    commands.spawn_bundle(PerspectiveCameraBundle {
+        transform: Transform::from_xyz(0.5, 1.0, -2.0)
+            .looking_at(Vec3::new(0.0, 0.3, 0.0), Vec3::Y),
+        ..Default::default()
+    });
+    const HALF_SIZE: f32 = 1.0;
+    commands.spawn_bundle(DirectionalLightBundle {
+        directional_light: DirectionalLight {
+            shadow_projection: OrthographicProjection {
+                left: -HALF_SIZE,
+                right: HALF_SIZE,
+                bottom: -HALF_SIZE,
+                top: HALF_SIZE,
+                near: -10.0 * HALF_SIZE,
+                far: 10.0 * HALF_SIZE,
+                ..Default::default()
+            },
+            shadows_enabled: true,
+            ..Default::default()
+        },
+        transform: Transform::from_xyz(-0.7, 1.7, -2.0)
+            .looking_at(Vec3::new(0.0, 0.3, 0.0), Vec3::Y),
+        ..Default::default()
+    });
+}
+
+fn spawn_and_print_gltf_objects(
+    mut commands: Commands,
+    mut done: ResMut<PrintDone>,
+    gltf_handle: Res<Handle<Gltf>>,
+    assets_gltf: Res<Assets<Gltf>>,
+    assets_gltfnode: Res<Assets<GltfNode>>,
+) {
+    if done.0 {
+        return;
+    }
+
+    // if the GLTF has loaded, we can navigate its contents
+    if let Some(gltf) = assets_gltf.get(gltf_handle.clone()) {
+        // Bevy allows you to easily spawn the entire scene.
+        // However since the Scene object here is a Bevy `Scene` object
+        // the mapping between Scene and GltfNodes is impossible to recover.
+        let scene_handle: &Handle<Scene> = &gltf.scenes[0];
+        commands.spawn_scene(scene_handle.clone());
+        // You can get a Vec of all top-level GltfNodes used in this scene
+        // through the `scene_to_nodes` map. This also enables you to recurse through the
+        // hierarchy by yourself should you need to do that.
+        let nodes: Vec<&GltfNode> = gltf.scene_to_nodes[scene_handle]
+            .iter()
+            .filter_map(|handle| assets_gltfnode.get(handle))
+            .collect::<Vec<_>>();
+
+        info!(
+            "The following nodes are currently being displayed {:#?}",
+            nodes
+        );
+        done.0 = true;
+    }
+}
+fn assert_correctness(
+    mut done: ResMut<AssertDone>,
+    gltf_handle: Res<Handle<Gltf>>,
+    assets_gltf: Res<Assets<Gltf>>,
+    assets_gltfnode: Res<Assets<GltfNode>>,
+) {
+    if done.0 {
+        return;
+    }
+    // if the GLTF has loaded, we can navigate its contents
+    if let Some(gltf) = assets_gltf.get(gltf_handle.clone()) {
+        let scene_handle: &Handle<Scene> = &gltf.scenes[0];
+        let nodes: Vec<&GltfNode> = gltf.scene_to_nodes[scene_handle]
+            .iter()
+            .filter_map(|handle| assets_gltfnode.get(handle))
+            .collect::<Vec<_>>();
+        assert_eq!(nodes.len(), 1);
+        assert_eq!(nodes[0].children[0].children.len(), 6);
+        done.0 = true;
+    }
+}
+#[derive(Default)]
+struct AssertDone(bool);
```

Comment:
we avoid assertions in tests as they don't show something that will be reused by users, could you remove this system?

Reply:
So, this was my suggestion. We could not run tests in the CI due to lingering difficulties around integration testing.

>  I tried to do that in b2caa7e however the event loop won't start as the tests aren't running on the main thread

I wanted *some* assurance that the feature was working and would continue to work.

I generally agree with your stance here, but I'm nervous about having subtle, untested code present.

Reply:
tests for that should live in https://github.com/bevyengine/bevy/blob/32f7997c566f4dcf8dc57b64f20193234256deb1/crates/bevy_gltf/src/loader.rs#L860-L861
It should be possible to split the scene part in an isolated function and test it, but complete asset loading is... not simple to test for now, though I didn't actually try.